### PR TITLE
wip: Argo d2

### PR DESCRIPTION
**Pull request by vitus133**
This is a proposed integration branch for argo cd day 2 operations.
It includes several PRs that we have in flight and will need a rebase after they are merged
/hold
cc @serngawy @nishant-parekh @imiller0

### CODE REVIEW

File: ztp/gitops-subscriptions/argocd/resource-hook-example/policygentemplates/common-ranGen.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
+---
```
Comment (imiller0) [inline ztp/gitops-subscriptions/argocd/resource-hook-example/policygentemplates/common-ranGen.yaml:1]: This is a duplicate of ztp/ztp-policy-generator/ranPolicyGenTempExamples/common-ranGen.yaml. We should have only one reference implementation and pull from there.

File: ztp/gitops-subscriptions/argocd/resource-hook-example/policygentemplates/kustomization.yaml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+resources:
```
Comment (imiller0) [inline ztp/gitops-subscriptions/argocd/resource-hook-example/policygentemplates/kustomization.yaml:1]: I'm not sure it makes sense to have a kustomization file here. These files are all committed to git where the argo pipeline is subscribed to them. We don't need a kustomization there.

File: ztp/resource-generator/src/watcher.py [hunk 3]
```diff
@@ -56,30 +56,34 @@ def watch_sites(self, rv):
...
+            cwd = os.path.join(
+                '/tmp/ztp/ztp-policy-generator',
+                'kustomize/plugin/policyGenerator/v1/policygenerator/')
+            command = [
+                './PolicyGenerator',
+                'dummy_arg',
+                paths[0],
+                template_dir,
+                paths[1],
+                'true', 'false', 'true']
```
Comment (imiller0) [inline ztp/resource-generator/src/watcher.py:81]: These booleans change between siteconfigs and policygentemplates. For policygentemplates it should be "true false false" and for siteconfigs it should be  true false true

File: ztp/ztp-policy-generator/ranPolicyGenTempExamples/kustomization.yaml [hunk 4]
```diff
@@ -0,0 +1,7 @@
+resources:
```
Comment (imiller0) [inline ztp/ztp-policy-generator/ranPolicyGenTempExamples/kustomization.yaml:1]: Is this kustomization file needed?

**Comment by nishant-parekh** [comment]
We need to add the `ztp:` prefix to the commit messages to make sure the CI does not fail on that.

**Comment by vitus133** [comment]
/close
Replaced by https://github.com/openshift-kni/cnf-features-deploy/pull/663

**Pull request closed without merging**